account-management-integration-tests/src/test/java/uk/gov/di/accountmanagement/api/MFAMethodsCreateHandlerIntegrationTest.java

```diff
@@ -71,24 +72,68 @@ void shouldReturn200AndMfaMethodDataWhenAuthAppUserAddsSmsMfa() {
 
         List<MFAMethod> mfaMethods = userStore.getMfaMethod(TEST_EMAIL);
 
-        assertTrue(
+        var retrievedSmsMethod =
                 mfaMethods.stream()
                         .filter(
                                 mfaMethod ->
                                         mfaMethod
                                                 .getMfaMethodType()
                                                 .equals(MFAMethodType.SMS.getValue()))
-                        .filter(
-                                mfaMethod ->
-                                        mfaMethod.getMfaIdentifier().equals(extractedMfaIdentifier))
-                        .filter(mfaMethod -> mfaMethod.getDestination().equals(TEST_PHONE_NUMBER))
+                        .findFirst()
+                        .get();
+
+        assertEquals(extractedMfaIdentifier, retrievedSmsMethod.getMfaIdentifier());
+        assertEquals(PriorityIdentifier.BACKUP.toString(), retrievedSmsMethod.getPriority());
+        assertEquals(TEST_PHONE_NUMBER, retrievedSmsMethod.getDestination());
+        assertTrue(retrievedSmsMethod.isEnabled());
+        assertTrue(retrievedSmsMethod.isMethodVerified());
+    }
+
+    @Test
+    void shouldReturn200AndMfaMethodDataWhenSmsUserAddsAuthAppMfa() {
+        userStore.addMfaMethodSupportingMultiple(
+                TEST_EMAIL,
+                MFAMethod.smsMfaMethod(
+                        true,
+                        true,
+                        TEST_PHONE_NUMBER,
+                        PriorityIdentifier.DEFAULT,
+                        UUID.randomUUID().toString()));
+        var response =
+                makeRequest(
+                        Optional.of(
+                                constructRequestBody(
+                                        PriorityIdentifier.BACKUP,
+                                        new AuthAppMfaDetail(
+                                                MFAMethodType.AUTH_APP, TEST_CREDENTIAL))),
+                        Collections.emptyMap(),
+                        Collections.emptyMap(),
+                        Map.of("publicSubjectId", TEST_PUBLIC_SUBJECT));
+        assertEquals(200, response.getStatusCode());
+
+        String extractedMfaIdentifier =
```

Comment:
It'd be good to assert against a full json response here (e.g. `assertEquals(response.getBody(), someJsonString`. We could potentially do this by grabbing the mfa identifier from the database first, and then chucking that in the json we're expecting. Just helps to have a readable json response when having a look at what the lambda does.